### PR TITLE
Add DRSDTCON to allow for convective DRSDT

### PR DESCRIPTION
**Pull request by totto82**
This adds a version of DRSDT useful for co2 storage. See the corresponding PR i opm-simulators for more details.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: src/opm/parser/eclipse/share/keywords/900_OPM/D/DRSDTCON [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  ],
+  "size": 1,
+  "items": [
+    {
+      "name": "DRSDT_MAX",
+      "value_type": "DOUBLE",
+      "dimension": "1"
+    },
+    {
+      "name": "Option",
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/share/keywords/900_OPM/D/DRSDTCON:14]: This should be capitalized plural: `OPTIONS`

File: src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp [hunk 2]
```diff
@@ -248,6 +248,18 @@ namespace {
         }
     }
 
+    void Schedule::handleDRSDTCON(const HandlerContext& handlerContext, const ParseContext&, ErrorGuard&) {
+        std::size_t numPvtRegions = this->m_static.m_runspec.tabdims().getNumPVTTables();
+        std::vector<double> max(numPvtRegions);
+        std::vector<std::string> options(numPvtRegions);
+        for (const auto& record : handlerContext.keyword) {
+            std::fill(max.begin(), max.end(), record.getItem("DRSDT_MAX").getSIDouble(0));
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp:256]: Use the compiled constants:
```C++
const auto& item = record.getItem<ParserKeywords::DRSDTCON::DRSDT_MAX>().getSIDouble(0);
```

File: src/opm/parser/eclipse/EclipseState/Schedule/OilVaporizationProperties.cpp [hunk 3]
```diff
@@ -91,6 +91,20 @@ namespace Opm {
         }
     }
 
+    void OilVaporizationProperties::updateDRSDTCON(OilVaporizationProperties& ovp, const std::vector<double>& maximums, const std::vector<std::string>& options){
+        ovp.m_type = OilVaporization::DRSDTCON;
+        ovp.m_maxDRSDT = maximums;
+        for (size_t pvtRegionIdx = 0; pvtRegionIdx < options.size(); ++pvtRegionIdx) {
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/Schedule/OilVaporizationProperties.cpp:97]: Seems like there is an assumption here that the number of elements in `options` array is equal to the range of `pvtRegionIdx`?

Comment (totto82) [reply src/opm/parser/eclipse/EclipseState/Schedule/OilVaporizationProperties.cpp:97]: This code is just a copy of the DRSDT code. The drsdt infrastructure supports region based drsdt using DRSDTR. For DRSDTCON and DRSDT the one given element is used for all PVT regions.

File: src/opm/parser/eclipse/share/keywords/900_OPM/D/DRSDTCON (hunk 1 above)
Comment (totto82) [reply src/opm/parser/eclipse/share/keywords/900_OPM/D/DRSDTCON:14]: OK. I will update the rest of the DRSDTs to make it consistent.

File: src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp (hunk 2 above)
Comment (totto82) [reply src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp:256]: Ok, I can update the rest of the file accordingly to make it consistent.

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**